lib/bridge.js

```diff
@@ -237,11 +240,109 @@ Bridge.prototype.run = function(port, config, appServiceInstance) {
         }
         self.opts.controller.onLog(line, false);
     });
+    this.customiseAppservice(this.appService);
     this.appService.listen(port);
     return this.loadDatabases();
 };
 
 /**
+ * Apply any customisations required on the appService object.
+ * Subclasses are intended to override this method where appropriate. If
+ * overridden the subclass method will still have to call the one in this
+ * class.
```

Comment:
OK; I've written an attempt at this in cc6a0a2.

I wonder if the function ought to handle the `/_matrix/app/:version` parts of the path itself though, to relieve call the callers from always having to remember to do it. Synapse has a similar structure, in which you even specify which API version(s) you support at the time you install the handler, so such version checking is done for you. This might look like (to pick some random example):

```
this.addAppServicePath({
    method: "GET",
    versions: ["r0", "r0.0.1"],
    path: "/thirdparty/throwAParty/:protocol",
    handler: function(req, res) { ... },
});
```

Generally, named arguments are nice, right?